tests/test_flirt.py

```diff
@@ -71,27 +97,31 @@ def test_called_from_mixed_functions(sample_038476):
     assert is_only_called_from_library_functions(sample_038476, func_addr) is True
 
 
-# call graph of 0x40B06C:
-#
-#       FUN_407BF0 <--- FUN_407B3C
-#           |              ^
-#           v              |
-#       FUN_408294         |
-#           |              |
-#           v              |
-#       FUN_40832F ---> FUN_4084D6
-#                 \     /
-#                  \   /
-#                   v v
-#              FUN_40868F
-#                    |
-#                    v
-#               FUN_408840
-#                    |
-#                    v
-#               FUN_40b06C
-
-
 def test_function_circular_call_graph(sample_038476):
     # this is a function whose call graph contains a cycle
+
+    # call graph of 0x40B06C:
+    #
+    # Note:
+    #   - recognized library functions are denoted by "(lib)"
+    #   - non-library functions have no designation
+    #
+    #       FUN_407BF0 <--- FUN_407B3C
+    #           |              ^
+    #           v              |
+    #       FUN_408294         |
+    #           |              |
+    #           v              |
+    #       FUN_40832F ---> FUN_4084D6
```

Comment:
there is no lib function in this path, are there examples where that is the case?

Reply:
Yes, there are no lib functions in this call graph. The testcase focues on testing a function whose call graph contains a cycle. In that case, `is_only_called_from_library_functions` should return false no matter if the cycle has lib funcs or not. We don't handle cyclic call graphs to avoid recursively calling for ever.

Reply:
assuming FUN_408840 would be a library function, can't we say then that FUN_40B06C is also library code?

Reply:
I don't think it is safe to make that assumption. I understand your point that a library function calling a user-defined function is not a common case (i.e. usually library functions call other library functions, and not custom code). But my only concern here is that if FUN_408840 is marked as a false-positive library function (via FLIRT) and given that assuming FUN_40B06C is a library function, would produce another false positive on top. 

What do you think?

Reply:
that's a danger of our entire approach here, we now make this assumption (independent of cyclic calls or not) or how do you see this differs?

Reply:
we could add further safe-guards ensuring that library functions are located close to each other / in chunks

e.g. there is an assumed start and end, like:

```
user-code
-----------
start library functions
...
end library functions
-----------
other data/code
```